http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/GrpcResource.java

```diff
@@ -20,6 +28,50 @@ public class GrpcResource {
     @GrpcClient("hello")
     Greeter client;
 
+    @Inject
+    @GrpcClient("reflection-service")
+    MutinyServerReflectionGrpc.MutinyServerReflectionStub reflection;
+
+    @GET
+    @Path("reflection/service/info")
+    @Produces(MediaType.APPLICATION_JSON)
+    public GrpcReflectionResponse reflectionServiceInfo() {
+        ServerReflectionRequest request = ServerReflectionRequest.newBuilder().setHost("localhost")
+                .setListServices("").build();
+        ServerReflectionResponse response = invoke(request);
+
+        int reflectionServiceCount = response.getListServicesResponse().getServiceCount();
+        List<String> reflectionServiceList = getReflectionServiceList(response);
+
+        return new GrpcReflectionResponse(reflectionServiceCount, reflectionServiceList);
+    }
+
+    @GET
+    @Path("reflection/descriptor/{protoFilename}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public byte[] reflectionFileDescriptor(@PathParam("protoFilename") String protoFilename) {
+        ServerReflectionRequest request = ServerReflectionRequest.newBuilder()
+                .setHost("localhost").setFileByFilename(protoFilename + ".proto").build();
+
+        ServerReflectionResponse response = invoke(request);
+        return response.getFileDescriptorResponse().toByteArray();
+    }
+
+    private ServerReflectionResponse invoke(ServerReflectionRequest request) {
+        return reflection.serverReflectionInfo(Multi.createFrom().item(request))
+                .collect().first()
+                .await().indefinitely();
```

Comment:
Wonderful opportunity to use reactive type :-) Please get rid of `.await().indefinitely()` and return `Multi<Byte[]>`. I don't see a reason to block inside the RESTEasy Reactive resource method.